components/server/src/workspace/workspace-service.ts

```diff
@@ -104,6 +104,8 @@ export class WorkspaceService {
         context: WorkspaceContext,
         normalizedContextURL: string,
     ): Promise<Workspace> {
+        await this.mayStartWorkspace(ctx, user, organizationId, this.db.findRegularRunningInstances(user.id));
```

Comment:
@geropl now looking at it, could it be reason why you did not move it here before because this code path also used from a prebuild manager?

Should we retest prebuilds too now?

Reply:
I tested Prebuilds seems to work, not sure whether there is something to watch out.

Reply:
> could it be reason why you did not move it here before because this code path also used from a prebuild manager?

That does not ring a bell. I think it was just too much code shifted around, and I was not sure "mayStartWorkspace" makes sense on "createWorkspace" (but I now think it does).

Also, from looking at the code, it checks all the invariants that should also hold for prebuilds. Only exception is that, because we select "find**Regular**RunningInstances(user.id)", it won't block on huge number of parallel prebuilds, but that matches current behavior, so it's fine. :+1: 
We could make the different more explicit, but don't have to do that now. :+1: